hrms/payroll/doctype/salary_slip/salary_slip.py

```diff
@@ -917,9 +917,7 @@ def get_amount_from_formula(self, struct_row, sub_period=1):
 		local_data = self.data.copy()
 		local_data.update({"start_date": start_date, "end_date": end_date, "posting_date": posting_date})
 
-		amount = self.eval_condition_and_formula(struct_row, local_data)
-
-		return amount
+		return flt(self.eval_condition_and_formula(struct_row, local_data))
```

Comment:
Instead of applying `flt` here, `eval_condition_and_formula` should take care of converting this with the row's precision. It's called at multiple places.